apps/developer-hub/src/components/EntropyTable/index.tsx

```diff
@@ -120,3 +129,80 @@ const getEntropyDeployments = async (
   const url = isMainnet ? FORTUNA_API_URLS.mainnet : FORTUNA_API_URLS.testnet;
   return await fetchEntropyDeployments(url);
 };
+
+function useEntropyFees(
+  chains: Record<string, EntropyDeployment>,
+): Record<string, bigint | undefined> {
+  const [feesByChain, setFeesByChain] = useState<
+    Record<string, bigint | undefined>
+  >({});
+  const clientsRef = useRef<Map<string, PublicClient>>(new Map());
+
+  function getClient(rpc: string): PublicClient {
+    const cached = clientsRef.current.get(rpc);
+    if (cached) return cached;
+    const client = createPublicClient({
+      transport: http(rpc, { timeout: 10_000, retryCount: 1 }),
+    });
+    clientsRef.current.set(rpc, client);
+    return client;
+  }
+
+  useEffect(() => {
+    const entries = Object.entries(chains);
+    if (entries.length === 0) return;
+
+    let isCancelled = false;
+
+    async function loadInBatches() {
+      const batchSize = 5;
+      for (let i = 0; i < entries.length; i += batchSize) {
+        if (isCancelled) return;
+        const batch = entries.slice(i, i + batchSize);
+        const results = await Promise.allSettled(
+          batch.map(async ([name, d]) => {
+            if (!d.rpc) return [name, undefined] as const;
+            try {
+              const client = getClient(d.rpc);
+              const fee = await client.readContract({
+                address: d.address as `0x${string}`,
+                abi: IEntropyV2 as unknown as Abi,
+                functionName: "getFeeV2",
+                args: [],
+              });
+              return [name, fee] as const;
+            } catch {
+              return [name, undefined] as const;
+            }
+          }),
+        );
+
+        const next: Record<string, bigint | undefined> = {};
+        for (const r of results) {
+          if (r.status === "fulfilled") {
+            const [name, fee] = r.value;
+            next[name] = fee as bigint;
```

Comment:
We should parse instead of typecasting, `zod` can do this or you can just do `if (typeof fee === "bigint")`